youtube_dl/extractor/bandlab.py

```diff
@@ -0,0 +1,172 @@
+                    'id': '91feeb36-8e10-4c91-ae57-ffac0a98c6b4',
+                    'title': 'How\'d I Lose You? (Intro)',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': 'd87c50a2-70cb-4937-9b97-3ae8646ca3aa',
+                    'title': 'Money $$$',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': 'ff2909ff-348f-448d-9d2c-7edbf2f0ec5e',
+                    'title': 'You\'ll Be Mine',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '53786c38-1f3c-4921-9271-793a64af7186',
+                    'title': 'Who You Are',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '0394bb27-113f-4d19-b902-f9c1fe6ba8a8',
+                    'title': 'In Your Eyes',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '2aa44689-c7fa-4d0e-b28a-e0d1dd570372',
+                    'title': 'The Same',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '281fe589-a559-4260-802d-78a6c7a973d8',
+                    'title': 'Fall In Love',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '83a9dc0a-702f-40fd-82f6-ab847f6a7b46',
+                    'title': 'Tried So Hard',
+                    'ext': 'm4a',
+                },
+            },
+            {
+                'info_dict': {
+                    'id': '3f6a4a1c-eb73-449f-bd45-f7958d6f2de1',
+                    'title': 'The End Of Everything',
+                    'ext': 'm4a',
+                },
+            }
+        ],
+        'info_dict': {
+            'id': 'dc26e307-e51f-ed11-95d7-002248452390',
+            'album': 'ENDLESS SUMMER',
+            'artist': 'Michael MacDonald'
+        },
+    },
+    {
+        'url': 'https://www.bandlab.com/hexatetrahedronx/collections/8fb1041c-e865-eb11-9889-0050f28a2802',
+        'playlist': [
+            {
+                'info_dict': {
+                    'id': '8f37e4aa-92c4-eb11-a7ad-0050f280467f',
+                    'title': 'psych ward',
+                    'ext': 'm4a'
+                }
+            }
+        ],
+        'info_dict': {
+            'id': '8fb1041c-e865-eb11-9889-0050f28a2802',
+            'album': 'DOOMTAPE',
+            'artist': '12days'
+        }
+    }]
+
+    def _real_extract(self, url):
+        resource_id = self._match_id(url)
+        kind_regex = re.compile(self._VALID_URL)
+        kind = kind_regex.match(url).group('kind')
```

Comment:
yt-dlp has the `_match_valid_url()` method, but (currently) we have to do this:
```suggestion
        resource_id, kind = re.match(self._VALID_URL).group('id', 'kind')
```